### PR TITLE
fix tensorrt test failed with pytorch 1.8+

### LINKED ISSUE #1454: tensorrt test failed
when trying to run test on test_tensorrt.py, there are many cuda errors.

My environment is cuda11+pytorch1.8.0+tensorrt7.1.3.4

I found out when forwarding  pytorch module with cpu mode, the error could be gone.

or remove the tensorrt part, the error could be gone.

so the issue is that when forwarding pytorch module with gpu mode and run tensorrt module together, the cuda error may be throw.

Here are the test log

[log.txt](https://github.com/open-mmlab/mmcv/files/7465778/log.txt)

### PR DESCRIPTION
**Pull request by twmht**
## Motivation

this PR resolves #1454

thanks the help from @grimoire

## Modification

add `-Xcompiler=-fno-gnu-unique`  in nvcc

### CODE REVIEW

**Review by grimoire — APPROVED** [review]
Cool! LGTM

**Comment by twmht** [comment]
Do I need to rerun the workflow? it seems that one of the failed ci is not related to this patch.

**Comment by zhouzaida** [comment]
> @grimoire
> 
> Do I need to rerun the workflow? it seems that one of the failed ci is not related to this patch.

Yet, I re-run it now.

File: setup.py [hunk 1]
```diff
@@ -161,6 +161,7 @@ def get_extensions():
         define_macros += [('MMCV_WITH_TRT', None)]
         cuda_args = os.getenv('MMCV_CUDA_ARGS')
         extra_compile_args['nvcc'] = [cuda_args] if cuda_args else []
+        extra_compile_args['nvcc'] += ['-Xcompiler=-fno-gnu-unique']
```
Comment (zhouzaida) [inline setup.py:164]: Hi @twmht , maybe we can add a comment here to explain why we need to add the flag

Comment (twmht) [reply setup.py:164]: sounds great. I would update the PR later.

Comment (twmht) [reply setup.py:166]: done

Comment (zhouzaida) [reply setup.py:166]: suggest merging the master branch to your branch which will fix the error when building documentation.

Comment (twmht) [reply setup.py:166]: you mean that I need to merge upstream master to this branch? I thought the unit test is only related to upstream master with this PR, am i wrong?

Comment (zhouzaida) [reply setup.py:166]: Yet, merge the upstream master to this branch.

**Review by zhouzaida — APPROVED** [review]

**Comment by twmht** [comment]
@grimoire  and @zhouzaida 

thanks for your help.

**Comment by zhouzaida** [comment]
> @grimoire and @zhouzaida
> 
> thanks for your help.

Thanks for your contribution and welcome to the OpenMMLab community.

**Review by ZwwWayne — APPROVED** [review]

**Pull request merged**